packages/expo-ui/src/swift-ui/Text/index.tsx

```diff
@@ -1,70 +1,76 @@
-import type { ColorValue } from 'react-native';
+import * as React from 'react';
 
-import { getTextFromChildren } from '../../utils';
 import { createViewModifierEventListener } from '../modifiers/utils';
 import { type CommonViewModifierProps } from '../types';
 
 export interface TextProps extends CommonViewModifierProps {
   /**
-   * The children of the text.
-   * Only string and number are supported.
+   * Text content or nested Text components.
    */
   children?: React.ReactNode;
-  /**
-   * The font weight of the text.
-   * Maps to iOS system font weights.
-   * @deprecated Use the font() modifier instead: modifiers={[font({ weight: 'bold' })]}
-   */
-  weight?:
-    | 'ultraLight'
-    | 'thin'
-    | 'light'
-    | 'regular'
-    | 'medium'
-    | 'semibold'
-    | 'bold'
-    | 'heavy'
-    | 'black';
-  /**
-   * The font design of the text.
-   * Maps to iOS system font designs.
-   * @deprecated Use the font() modifier instead: modifiers={[font({ design: 'rounded' })]}
-   */
-  design?: 'default' | 'rounded' | 'serif' | 'monospaced';
-  /**
-   * The font size of the text.
-   * @deprecated Use the font() modifier instead: modifiers={[font({ size: 18 })]}
-   */
-  size?: number;
-  /**
-   * The line limit of the text.
-   */
-  lineLimit?: number;
-  /**
-   * The color of the text.
-   */
-  color?: ColorValue;
 }
 
-type NativeTextProps = Omit<TextProps, 'children'> & {
-  text: string;
+type NativeTextProps = CommonViewModifierProps & {
+  text?: string;
+  children?: React.ReactNode;
 };
 
-const TextNativeView: React.ComponentType<Omit<TextProps, 'children'> & { text: string }> =
-  requireNativeView('ExpoUI', 'TextView');
+const TextNativeView: React.ComponentType<NativeTextProps> = requireNativeView(
+  'ExpoUI',
+  'TextView'
+);
 
-function transformTextProps(props: TextProps): NativeTextProps {
-  const { children, modifiers, ...restProps } = props;
-  const text = getTextFromChildren(children);
-  return {
-    modifiers,
-    ...(modifiers ? createViewModifierEventListener(modifiers) : undefined),
-    ...restProps,
-    text: text ?? '',
-  };
+function transformChildren(children: React.ReactNode): React.ReactNode[] | null {
+  const childArray = React.Children.toArray(children);
+  if (childArray.length === 0) return null;
+
+  const result: React.ReactNode[] = [];
+  let keyIndex = 0;
+
+  for (const child of childArray) {
+    if (typeof child === 'string' || typeof child === 'number') {
+      result.push(<TextNativeView key={`text-${keyIndex++}`} text={String(child)} />);
+    } else if (React.isValidElement(child) && child.type === Text) {
+      result.push(child);
+    }
+  }
+
+  return result.length > 0 ? result : null;
 }
 
 export function Text(props: TextProps) {
-  return <TextNativeView {...transformTextProps(props)} />;
+  const { children, modifiers, ...restProps } = props;
+
+  if (!children) {
+    return null;
+  }
+
+  const childArray = React.Children.toArray(children);
```

Comment:
we call `React.children.toArray` here and inside `transformChildren`. any chance we can only call once?

maybe merge isSimpleText logic into `transformChildren`?